sdks/smart-wallet-sdk/src/smartWallet.ts

```diff
@@ -60,16 +80,16 @@ export class SmartWallet {
    * Get the mode type from the options
    */
   public static getModeFromOptions(options: ExecuteOptions): ModeType {
-    if(options.revertOnFailure) {
-      return ModeType.BATCHED_CALL_CAN_REVERT
+    if(options.shouldRevert) {
+      return ModeType.BATCHED_CALL;
     }
 
-    return ModeType.BATCHED_CALL
+    return ModeType.BATCHED_CALL_CAN_REVERT
   }
 
   /** Internal methods */
   
-  protected static _encodeExecute(mode: ModeType, data: `0x${string}`): `0x${string}` {
+  protected static _encodeERC7821Execute(mode: ModeType, data: `0x${string}`): `0x${string}` {
```

Comment:
imo we should be encoding calldata as a direct call to execute not through the 7821 wrapper!

Reply:
this is only supported for the legacy mode if desired, I marked it as deprecated do you think we should just remove it all together?

Reply:
I mean I can't see our internal code using this codepath but I think its fair to support in the sdk !